pkg/nfd-gc/nfd-gc.go

```diff
@@ -126,20 +153,40 @@ func (n *nfdGarbageCollector) garbageCollect() {
 		nodeNames.Insert(node.Name)
 	}
 
-	nrts, err := n.topoClient.TopologyV1alpha2().NodeResourceTopologies().List(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		klog.ErrorS(err, "failed to list NodeResourceTopology objects")
-		return
+	// Handle NodeFeature objects
+	nfs, err := n.nfdClient.NfdV1alpha1().NodeFeatures("").List(context.TODO(), metav1.ListOptions{})
+	if errors.IsNotFound(err) {
+		klog.V(2).InfoS("NodeFeature CRD does not exist")
```

Comment:
nit: Perhaps should be CR instead of CRD?

Reply:
In this case it is CRD indeed. If you cannot list the resource (and apiserver returns notfound) it means the whole api resource type does not exist